src/line_ending.rs

```diff
@@ -0,0 +1,50 @@
+//! TODO
+
+use std::str::FromStr;
+
+/// TODO doc
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub enum LineEnding {
+    /// TODO
+    CR,
+    /// TODO
+    CRLF,
+    /// TODO
+    LF,
+}
+
+impl LineEnding {
+    /// TODO
+    #[inline]
+    pub const fn len_chars(&self) -> usize {
+        match self {
+            Self::CRLF => 2,
+            _ => 1,
+        }
+    }
+
+    /// TODO
+    #[inline]
+    pub const fn as_str(&self) -> &'static str {
+        match self {
+            Self::CRLF => "\u{000D}\u{000A}",
+            Self::LF => "\u{000A}",
```

Comment:
I'm not very good at remembering the hexadecimal ASCII values :-) Could we use `\n` instead here and above?